datacube/testutils/io.py

```diff
@@ -132,32 +137,168 @@ def native_geobox(ds: Dataset, measurements=None, basis: str | None = None):
+
+
+def _native_load_1(
+    sources: xr.DataArray,
+    bands: tuple[str, ...],
+    *,
+    dst_geobox: GeoBox | None = None,
+    optional_bands: tuple[str, ...] | None = None,
+    basis: str | None = None,
+    pad: int | None = None,
+    **kw,
+) -> xr.Dataset:
+    """Load datasets with native crs
+    :param sources: grouped datasets
+    :param bands: List of band names to load
+    :param dst_geobox: Geobox of final output, if None then use geobox of input dataset
+    :param optional_bands: List of optional band names to load
     :param basis: Name of the band to use for computing reference frame, other
     bands might be reprojected if they use different pixel grid
+    :param pad: number of pixels to pad the geobox
 
-    :param kw: Any other parameter load_data accepts
+    :param kw: Any other parameter that load_data accepts
 
     :return: Xarray dataset
     """
-    from datacube import Datacube
-
-    geobox = native_geobox(
-        ds, measurements, basis
-    )  # early exit via exception if no compatible grid exists
-    mm = (
-        ds.product.measurements
-        if measurements is None
-        else ds.product.lookup_measurements(measurements)
-    )
+    if basis is None:
+        basis = bands[0]
+    (ds,) = sources.data[0]
+    load_geobox = compute_native_load_geobox(ds, basis, dst_geobox)
+    if pad is not None:
+        load_geobox = gbox_pad(load_geobox, pad)
+
+    mm = ds.product.lookup_measurements(bands)
+    if optional_bands is not None:
+        for ob in optional_bands:
+            try:
+                om = ds.product.lookup_measurements(ob)
+            except KeyError:
+                continue
+            else:
+                mm.update(om)
+
+    xx = Datacube.load_data(sources, load_geobox, mm, **kw)
+    return xx
+
+
+def native_load(
+    dss: Sequence[Dataset],
+    bands: Sequence[str],
+    groupby: Callable[..., Any],
+    *args,
+    dst_geobox: GeoBox | None = None,
+    optional_bands: tuple[str, ...] | None = None,
+    basis: str | None = None,
+    pad: int | None = None,
+    **kw,
+):
+    """Load datasets in native resolution.
 
-    return Datacube.load_data(
-        Datacube.group_datasets([ds], "time"), geobox, measurements=mm, **kw
-    )
+    :param dss: Datasets
+    :param bands: List of band names to load
+    :param groupby: Function to group the datasets
+    :param args: positional passed into gropuby
+
+    :param dst_geobox: Geobox of final output, if None then use geobox of input dataset
+    :param optional_bands: List of optional band names to load
+    :param basis: Name of the band to use for computing reference frame, other
+    bands might be reprojected if they use different pixel grid
+    :param pad: number of pixels to pad the geobox
+
+    :param kw: Any other parameter groupby or _native_load_1 accepts
+
+    :return: Xarray dataset or generator of Xarray dataset
+    """
+    # Filter **kwargs to match what op accepts
```

Comment:
The parameter is called `**kw`.

What does `op` mean/refer to?

Reply:
Appears to mean the `groupby` function - which is why its type signature has to be left vague above.

(Docstring could definitely be clearer though)